docs/en/operations/settings/settings.md

```diff
@@ -4384,6 +4384,17 @@ Possible values:
 
 Default value: `ignore`.
 
+## first_day_of_week
+
+The first day of the week assumed by [`toStartOfInterval`](../../sql-reference/functions/date-time-functions.md#toStartOfInterval) function when using weeks as unit.
+
+Possible values:
+
+- Monday - Week starts on Monday
+- Sunday - Week starts on Sunday
```

Comment:
Why not allow Saturday and Friday? There are cases for that.

Reply:
Why not extend it to `toStartOfWeek` then?

Reply:
It is way too dangerous to have it as a setting - see the comment about indices and distributed queries.

Reply:
Let's revert this if it could be dangerous and will contribute to confusion instead of helping.

We can keep track of the different starting days in different functions and improve the documentation being explicit about which day is used by default in each function.

Reply:
> It is way too dangerous to have it as a setting - see the comment about indices and distributed queries.

@alexey-milovidov could you please elaborate on why this is dangerous? Or point me towards some literature about it? I have a similar solution in mind for a related issue, so I'd like to know more. Thanks.